theano/tensor/opt.py

```diff
@@ -4398,6 +4397,17 @@ def local_mul_to_sqr(node):
             if node.inputs[0] is node.inputs[1]:
                 return [T.sqr(node.inputs[0])]
 
+@register_specialize
+@gof.local_optimizer([T.int_div, T.floor_div])
+def local_div_by_one(node):
+    """x // 1 -> x
+    """
+    if node.op in [T.int_div]:
+        if len(node.inputs) == 2:
```

Comment:
I'm 95% certain that the fact that the scalar_op is an IntDiv will guarantee that the node has exactly 2 inputs.